### ISSUE TITLE
Dimensionless_coagulation_kernel_parameterized outputs imaginary numbers

### ISSUE THREAD

**Issue by wkchuang** [uncscode/particula#81]
Is this working as intended?
Reproduce this by adding a print statement for the answer and calling 
`pytest -s particle_test.py`.
I get `(0.0035952570681724756+1.2816299634238463e-74j) dimensionless` for our 
current tests. 
The imaginary numbers carry through to `dimensioned_coagulation_kernel`.

**Comment by ngam** [comment]
Can you make sure you have the very latest updates?

**Comment by ngam** [comment]
This is most likely due to a constraint that was deleted earlier for some reason. In the earliest versions, I used to have a hard-coded statement that would take care of this, but I believe someone took it out. However, the issue is really about using the  constraint correctly. For example see around these code snippets.

https://github.com/uncscode/particula/blob/6ecb6a1a3a4b8e56c1a5b0e1f94f38638c64ca86/particula/aerosol_dynamics/particle.py#L393-L402

https://github.com/uncscode/particula/blob/6ecb6a1a3a4b8e56c1a5b0e1f94f38638c64ca86/particula/aerosol_dynamics/particle.py#L415-L422

Basically, you'd want to make sure `self.coulomb_potential_ratio` is constraint to what the `authors` parameterized them to be. In my latest edits, I thought I took care of all of that by adding all constraints, but I could I have missed something...

**Comment by ngam** [comment]
Anyway, when I run the tests, I don't get imaginary stuff --- maybe you still have earlier versions locally?

**Comment by ngam** [comment]
Also, note: I gathered all tests under a test folder due to practical purposes. Essentially, we want to make sure we are not packaging the tests as part of the package itself; so as an intermediate solution, I gathered them in their own separate module within the package. The idea is that they could be run (not really without pytest though) after installing the package. Ultimately, I would want to take them all the way out and organize them in a better way.

**Comment by ngam** [comment]
Let me know if you can replicate the errors using something like this in a notebook standalone (colab, etc.): https://uncscode.github.io/particula/tutorials/001_ParticleClass.html

**Comment by ngam** [comment]
Any update on this @wkchuang?

**Comment by wkchuang** [comment]
I need to replicate it in the standalone notebook, but I'm still seeing it locally

**Comment by ngam** [comment]
Can you send me the exact call? I will try to replicate it myself

**Comment by wkchuang** [comment]
What do you need?

1. Insert a print statement in test_dimensionless_coagulation_kernel_parameterized, e.g.
`print(small_particle.dimensionless_coagulation_kernel_parameterized(
        large_particle, standard_environment
    ))`
2. run pytest -s particle_test.py
The output of the console should print out the result.

**Comment by ngam** [comment]
The dimensioned one is okay: `.......4.670176867752661e-13 meter ** 3 / second`

**Comment by ngam** [comment]
Anyway, this is due to the the like--like coagulation.

**Comment by wkchuang** [comment]
Ok, so the imaginary part is accurate?

**Comment by ngam** [comment]
| `small_particle.charge()` | `large_particle.charge()` | `dimensionless_coagulation_kernel_parameterized( arge_particle, standard_environment)` |
| --- | --- | --- | 
| `1` | `-1` | `0.001499926972164872` | 
| `1` | `0` | `0.002129567040935978` | 
| `1` | `1` | `0.0035952570681724756+1.2816299634238463e-74j` | 

> Ok, so the imaginary part is accurate?

nope. I will need to go through the steps again to make sure. I will resolve it. When the potential ratio is a certain sign, it should skip the calculation and default to something else. I will trace this this weekend.

**Comment by wkchuang** [comment]
cool, thanks for following up on that!

**Comment by ngam** [comment]
https://github.com/uncscode/particula/blob/719080530e79342c7842da56c3d496b573d4b342/particula/aerosol_dynamics/particle.py#L393-L402

Most likely I need to uncomment the control flow here

**Comment by ngam** [comment]
These parameterizations basically only apply for certain conditions, for example: `coulomb_potential_ratio` negative/positive. If not this, then, something more complex is going on. I will review this again this weekend.

**Comment by ngam** [comment]
keep

**Comment by ngam** [comment]
I have disabled the calculations that produce ths particular issue. Once reenabled, likely in https://github.com/uncscode/particula/milestone/6, I will make sure we don't have these issues.
